src/nvim/path.c

```diff
@@ -1098,7 +1102,7 @@ int gen_expand_wildcards(int num_pat, char_u **pat, int *num_file,
     p = pat[i];
 
     if (vim_backtick(p))
-      add_pat = expand_backtick(&ga, p, flags);
+      add_pat = (int)expand_backtick(&ga, p, flags);
```

Comment:
Have you checked that `expand_backtick` won't return a value more than INT_MAX? Normally an assert statement would be used here, although I might be missing something.

This goes for the other instances of integer casting in this PR.